pandas/core/arrays/integer.py

```diff
@@ -94,10 +94,14 @@ def construct_array_type(cls) -> Type["IntegerArray"]:
 
     def _get_common_dtype(self, dtypes: List[DtypeObj]) -> Optional[DtypeObj]:
         # for now only handle other integer types
-        if not all(isinstance(t, _IntegerDtype) for t in dtypes):
+        if not all(
+            isinstance(t, _IntegerDtype)
+            or (isinstance(t, np.dtype) and np.issubdtype(t, np.integer))
+            for t in dtypes
+        ):
```

Comment:
I added the ability to concat nullable int + numpy int, to be able to test the specific frame case. 
But still need to do this properly + test it (see also https://github.com/pandas-dev/pandas/issues/34095)